lib/class-wp-theme-json-gutenberg.php

```diff
@@ -301,7 +301,9 @@ public function __construct( $theme_json = array(), $origin = 'theme' ) {
 				$path   = array_merge( $node['path'], $preset_metadata['path'] );
 				$preset = _wp_array_get( $this->theme_json, $path, null );
 				if ( null !== $preset ) {
-					_wp_array_set( $this->theme_json, $path, array( $origin => $preset ) );
+					if ( 'user' !== $origin || isset( $preset[0] ) ) {
```

Comment:
I'd think we want to rename the `user` origin to `custom`, as we did with `core` to `default` to match the UI, class/css var generation, etc. Perhaps we can do this in a different PR that lands before this one?

Reply:
What's the use case `isset( $preset[ 0 ] )` is checking for? My understanding is that we should always skip reorganizing the presets if the origin is the user, as we'll store them already keyed by origin, so this could be simplified to: `if ( null != $preset && 'user' !== $origin )`.

Additionally, I wonder if we need to keep back compatibility with existing user data in the database. We could add an extra step to normalize the user data by origin for that case.